pymeasure/instruments/agilent/agilentN9320A.py

```diff
@@ -0,0 +1,262 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2019 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_range, strict_discrete_set
+
+from time import sleep
+import numpy as np
+import pandas as pd
+
+
+
+class AgilentN9320A(Instrument):
+    """ Represents the AgilentN9320A Spectrum Analyzer
+    and provides a high-level interface for taking scans of
+    high-frequency spectrums
+    """
+
+    FREQ_LIMIT = [9e3, 3.08e9] #Frequency limit in Hz
+    SPAN_LIMIT = [0, 3e9]      #In zero span the X axis represents time
+    REF_LIMIT = [-100, 50]     #Reference level limit in dBm
+    ATT_LIMIT = [0, 70]        #Input attenuator limit in dB
+    RES_LIMIT = [10, 30, 100, 300, 1000, 3000,
+                10000, 30000, 100000, 300000]      #RBW values
+    VID_LIMIT = RES_LIMIT       #VBW values
+    PEAK_TH = -70.             #Peak threshold for peak recognition in dBm
+    PEAK_EXC = 3               #Peak excursion
+    D_FACTOR = 2               #A multiplicative factor for the sweep time in
+                               #peak function
+    DELAY = 0.2                #A delay in second can be useful
+
+
+    start_frequency = Instrument.control(
+        ":SENS:FREQ:STAR?", ":SENS:FREQ:STAR %e Hz",
+        """ A floating point property that represents the start frequency
+        in Hz. This property can be set.
+        """,
+		validator=strict_range,
```

Comment:
Please correct the indentation here to be consistent. I would suggest that the docstring ends on the previous line:
```
    in Hz. This property can be set. """,
    validator=strict_range,
    values=FREQ_LIMIT,
)
```